src/java.base/share/classes/java/lang/String.java

```diff
@@ -3218,14 +3218,52 @@ public String replace(CharSequence target, CharSequence replacement) {
      * @since 1.8
      */
     public static String join(CharSequence delimiter, CharSequence... elements) {
-        Objects.requireNonNull(delimiter);
-        Objects.requireNonNull(elements);
-        // Number of elements not likely worth Arrays.stream overhead.
-        StringJoiner joiner = new StringJoiner(delimiter);
-        for (CharSequence cs: elements) {
-            joiner.add(cs);
+        var delim = delimiter.toString();
+        var elems = new String[elements.length];
+        for (int i = 0; i < elements.length; i++) {
+            elems[i] = String.valueOf(elements[i]);
+        }
+        return join("", "", delim, elems, elems.length);
+    }
+
+    /**
+     * Designated join routine.
```

Comment:
Did you mean "dedicated"?

Reply:
No, I meant designated. It is the routine that all other public API entry points call at the end to do the job. Would some other word more accurately describe that? I definitely didn't mean "dedicated".

Reply:
Oh then sorry, I thought it was a typo of some sort. I'd have said something like "Centralized join logic". But whatever works for you.